scipy/stats/_multivariate.py

```diff
@@ -3850,3 +3851,418 @@ def rvs(self, dim, size=1, random_state=None):
+        %(_doc_default_callparams)s
+
+        Returns
+        -------
+        cov : array_like
+            The covariance matrix of the distribution
+        """
+        M, m, n, mncond = self._process_parameters(m, n)
+        M, n = M[..., np.newaxis], n[..., np.newaxis]
+        output = n * m/M * (M-m)/M * (M-n)/(M-1)
+        if output.shape[-1] == 1:
+            self._checkresult(output, mncond, np.NAN).squeeze(-1)
+        return self._checkresult(output, mncond, np.NAN)
+
+    def cov(self, m, n):
+        """
+        Covariance matrix of the Multivariate Hypergeometric distribution.
+
+        Parameters
+        ----------
+        %(_doc_default_callparams)s
+
+        Returns
+        -------
+        cov : array_like
+            The covariance matrix of the distribution
+        """
+        # cov( x_i,x_j ) = -n * (M-n)/(M-1) * (K_i*K_j) / (M**2)
+        M, m, n, mncond = self._process_parameters(m, n)
+        M = M[..., np.newaxis, np.newaxis]
+        n = n[..., np.newaxis, np.newaxis]
+        output = (-n * (M-n)/(M-1) / (M**2) *
+                  np.einsum("...i,...j->...ij", m, m))
+        dim = m.shape[-1]
+        M, n = M[..., 0, 0], n[..., 0, 0]
+        # diagonal entries need to be computed differently
+        for i in range(dim):
+            output[..., i, i] = (n * (M-n)/(M-1) *
+                                 m[..., i]*(M-m[..., i]) / (M**2))
+
+        return self._checkresult(output, mncond, np.NAN)
+
+    def rvs(self, m, n, size=1, random_state=None):
+        """
+        Draw random samples from a Multivariate Hypergeometric distribution.
+
+        Parameters
+        ----------
+        %(_doc_default_callparams)s
+        size : integer or iterable of integers, optional
+            Number of samples to draw (default 1).
+        %(_doc_random_state)s
+
+        Returns
+        -------
+        rvs : array_like
+            Random variates of shape (`size`, `len(p)`)
+
+        Notes
+        -----
+        %(_doc_callparams_note)s
+        """
+        M, m, n, mncond = self._process_parameters(m, n)
+
+        random_state = self._get_random_state(random_state)
+        try:
+            # Generator is only available in numpy >= 1.17
+            if isinstance(random_state, np.random.Generator):
+                out = random_state.multivariate_hypergeometric(m, n,
+                                                               size)
+                return _squeeze_output(out)
+        except AttributeError:
+            pass
+
+        if isinstance(size, int):
+            size = (size, )
+
+        rvs = np.empty(size + (m.shape[-1], ), dtype=m.dtype)
+        rem = M
+        _, rem, n = np.broadcast_arrays(rvs[..., 0], rem, n)
+
+        for c in range(m.shape[-1] - 1):
+            rem = rem - m[..., c]
+            rvs[..., c] = ((n != 0) *
+                            random_state.hypergeometric(m[..., c], rem,
+                                                        n + (n == 0),
```

Comment:
A few lines here are a "workaround" for sampling from NumPy's legacy hypergeometric distribution. This distribution doesn't allow to sample when `n=0`. This is a problem since the newer numpy distributions allow it and marginal method in numpy uses that distribution to sample from the multivariate hypergeometric distribution. Hence, this workaround just masks the output with 0 when `n=0` and to avoid the error, it just changes the value of n to 1 which will not affect the overall sampling

Reply:
In fact, I have also added a test for `rvs` method that passes and shows that the sampling is unbiased.